ipykernel/kernelbase.py

```diff
@@ -892,20 +892,25 @@ def _send_interupt_children(self):
             # Prefer process-group over process
             # but only if the kernel is the leader of the process group
             if pgid and pgid == pid and hasattr(os, "killpg"):
-                try:
-                    os.killpg(pgid, SIGINT)
-                    return
-                except OSError:
-                    pass
-            try:
+                os.killpg(pgid, SIGINT)
+            else:
                 os.kill(pid, SIGINT)
-            except OSError:
-                pass
```

Comment:
move out OSError try except

Reply:
This is a change in behavior.  I think we want to do something like:

```python
try:
    os.killpg(pgid, SIGINT)
except OSError:
    os.kill(pid, SIGINT)
    raise
os.kill(pid, SIGINT)
````

Reply:
oops youre right